src/dialects/mysql/query-generator-typescript.ts

```diff
@@ -43,4 +60,50 @@ export class MySqlQueryGeneratorTypeScript extends AbstractQueryGenerator {
 
     return `DROP INDEX ${this.quoteIdentifier(indexName)} ON ${this.quoteTable(tableName)}`;
   }
+
+  /**
+   * Generates an SQL query that returns the foreign key constraint of a given column.
+   *
+   * @param   tableName  The table or associated model.
+   * @param   columnName The name of the column.
+   * @returns            The generated SQL query.
+   */
+  getForeignKeyQuery(tableName: TableNameOrModel, columnName: string) {
+    const table = this.extractTableDetails(tableName);
+
+    const quotedTableName = this.escape(table.tableName);
+    const quotedSchemaName = this.escape(table.schema!);
+    const quotedColumnName = this.escape(columnName);
+
+    return joinSQLFragments([
+      GET_FOREIGN_KEYS_PREFIX,
+      'WHERE (',
+      `REFERENCED_TABLE_NAME = ${quotedTableName}`,
+      `AND REFERENCED_TABLE_SCHEMA = ${quotedSchemaName}`,
+      `AND REFERENCED_COLUMN_NAME = ${quotedColumnName}`,
```

Comment:
this doesn't sound right. It's selecting foreign keys that reference this table & column, instead of the foreign key constraint *of* the column/constraint, which seems to be what all other dialects are doing